index.js

```diff
@@ -119,6 +119,43 @@ const cachedElmCompiler = () => {
   return { cache, compileToStringSync };
 };
 
+const fileExists = (file) => {
+  return fs.stat(file).then(stat => stat.isFile()).catch(_ => false);
+};
+
+// Attempts to resolve a file path by joining to each load path, and returns the
+// resolved path if that file exists.
+// If no load paths are provided, or none resolve, the file path is assumed to
+// be relative to `resolveDir`.
+const resolvePath = async (resolveDir, filePath, loadPaths = []) => {
+  const relativePath = path.join(resolveDir, filePath);
+
+  if (loadPaths.length === 0) {
```

Comment:
Maybe we should push the `resolveDir` to `loadPaths` so the relative path behavior is preserved?

Reply:
The only problem I can see with this approach is that it would require an extra filesystem check here https://github.com/phenax/esbuild-plugin-elm/pull/24/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R140
while using the guard condition the code would avoid it. 
This might slightly slow down the build especially when no loadPaths are passed through
However the code should be easier to follow e.g.

```
const resolvePath = async (resolveDir, filePath, loadPaths = []) => {
  const relativePath = path.join(resolveDir, filePath);
  const allLoadPaths = loadPaths.concat([relativePath])

  for (let loadPath of allLoadPaths) {
    let joinedPath = path.join(loadPath, filePath);

    if (await fileExists(joinedPath)) {
      return joinedPath;
    }
  }

  return relativePath;
};
```

Reply:
The problem I see with the current implementation is that relative paths won't work when `source-directories` is specified which would be a breaking change.

I think we can go with this instead then so we avoid the extra check -
```js
const resolvePath = async (resolveDir, filePath, loadPaths = []) => {
  for (const loadPath of loadPaths) {
    const joinedPath = path.join(loadPath, filePath);

    if (await fileExists(joinedPath)) {
      return joinedPath;
    }
  }

  return path.join(resolveDir, filePath);
};
```

Reply:
I see, makes sense thanks! I updated the pr